lib/policies/proxy/proxy.js

```diff
@@ -51,15 +54,49 @@ module.exports = function (params, config) {
   if (endpoint.url) {
     return function proxyHandler (req, res) {
       logger.debug(`proxying to ${endpoint.url}, ${req.method} ${req.url}`);
-      proxy.web(req, res, { target: endpoint.url });
+      proxy.web(req, res, proxyOptions);
     };
   }
 
   // multiple urls will load balance, defaulting to round-robin
   const strategy = params.strategy || 'round-robin';
-  const balancer = createStrategy(strategy, proxy, endpoint.urls);
+  const balancer = createStrategy(strategy, proxy, proxyOptions, endpoint.urls);
 
   return function balancerProxyHandler (req, res) {
     return balancer.proxy(req, res);
   };
 };
+
+// Parse endpoint URL if single URL is provided.
+// Extend proxy options by allowing and parsing keyFile and certFile.
+function parseProxyOptions (proxyOptions = {}, endpoint = {}) {
+  let parsedProxyOptions = {};
+
+  if (Object.keys(endpoint).includes('proxyOptions')) {
+    parsedProxyOptions = Object.assign(endpoint.proxyOptions, proxyOptions);
+  }
+
+  if (!Object.keys(parsedProxyOptions).includes('target')) parsedProxyOptions.target = {};
```

Comment:
The point of checking those properties is know whether we need to to perform a read from the corresponding file path. If we don't, we are always assuming keyFile will be set.

Reply:
I agree with that — I probably commented on the wrong line.

What I'm trying to say is that if this condition is true:

` if (!Object.keys(parsedProxyOptions).includes('target')) parsedProxyOptions.target = {};`

Then `target` is an empty object for sure. Therefore it does not make sense to make the checks you're doing in the following line such as:

`if (parsedProxyOptions.target.keyFile) {…}`
because target is a empty object for sure.

Reply:
Ahah! Yes. Of course! I'll amend.